fastlane/actions/sign_apk.rb

```diff
@@ -0,0 +1,94 @@
+module Fastlane
+  module Actions
+    module SharedValues
+      SIGNED_APK_PATH = :SIGNED_APK_PATH
+    end
+
+    class SignApkAction < Action
+      def self.run(params)
+      
+      UI.user_error!("Couldn't find '*release-unsigned.apk' file at path 'app/build/outputs/apk/'") unless params[:apk_path]
+
+      UI.user_error("Need keystore in order to sign apk") unless params[:keystore_path]
+
+      sign_cmd = ["jarsigner -verbose -sigalg SHA1withRSA -digestalg SHA1"]
+      sign_cmd << ["-keystore #{params[:keystore_path].shellescape}" ] if params[:keystore_path]
+      sign_cmd << ["#{params[:apk_path].shellescape}"] if params[:apk_path]
+      sign_cmd << ["'#{params[:alias]}'"] if params[:alias]
+      sign_cmd << ["-keypass #{params[:keypass] ? params[:keypass] : params[:storepass]}"] if params[:keypass] || params[:storepass]
+      sign_cmd << ["-storepass #{params[:storepass]}"] if params[:storepass]
+      sign_cmd << ["-tsa #{params[:tsa]}"] if params[:tsa]
+
+      if params[:signed_apk_path]
+        sign_cmd << ["-signedjar #{params[:signed_apk_path]}" ]
+        Actions.lane_context[SharedValues::SIGNED_APK_PATH] = "#{params[:signed_apk_path]}"
+      elsif params[:apk_path].include?("unsigned")
+        sign_cmd << ["-signedjar #{params[:apk_path].gsub('-unsigned', '')}"]
+        Actions.lane_context[SharedValues::SIGNED_APK_PATH] = "#{params[:apk_path].gsub('-unsigned', '')}"
+      end
+
+      Fastlane::Actions.sh(sign_cmd, log: true)
+    end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        "Sign a Android apk with a java keystore"
+      end
+
+      def self.available_options 
+
+        apk_path_default = Dir["*.apk"].last || Dir[File.join("app", "build", "outputs", "apk", "*release-unsigned.apk")].last
+
+        [
+          FastlaneCore::ConfigItem.new(key: :apk_path,
+                                       env_name: "apk_path",
```

Comment:
yes, the key, sagebio.jks.enc, is encrypted and committed in the root of this project.  it is decrypted on the build machine to sign the apk.  The passwords for the key file are all encrypted travis env vars.